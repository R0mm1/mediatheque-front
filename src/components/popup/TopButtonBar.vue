<template>
    <div id="med_popup_topbuttonbar">
        <InputButton v-for="button in buttons" :key="button.tabName"
                     v-on:click.native="setTab(button.tabName)"
                     :label-custom-classes="button.iconClassName"
                     :custom-classes="cIsSelected(button.tabName)">
        </InputButton>

        <InputButton v-on:click.native="$emit('popup-wanna-close')"
                     label-custom-classes="fas fa-times"></InputButton>
    </div>
</template>

<script>
    import InputButton from "../form/elements/InputButton";

    export default {
        name: "TopButtonBar",
        components: {InputButton},
        props: {
            buttons: {type: Array, default: () => []},
            defaultTab: {type: String, required: true}
        },
        data() {
            return {
                selectedTabName: null
            }
        },
        methods: {
            setTab(tabName) {
                this.selectedTabName = tabName;
                this.$emit('tab-button-clicked', tabName);
            },
            cIsSelected(tabName) {
                return (this.selectedTabName === null && tabName === this.defaultTab) || (tabName === this.selectedTabName) ? ['selected'] : [];
            }
        }
    }

    const TopButtonBarElement = function (iconClassName, tabName) {
        this.iconClassName = iconClassName;
        this.tabName = tabName;
    };

    export {TopButtonBarElement};
</script>

<style lang="scss">
    @import "../../assets/scss/colors";

    #med_popup_topbuttonbar {
        display: flex;

        > .form_element {
            width: 4rem;
            margin: 0;
            display: flex;
            flex-direction: column;
            text-align: center;
            justify-content: center;

            &.selected:after {
                content: ' ';
                display: block;
                background-color: $shade1;
                width: 100%;
                height: 3px;
                position: absolute;
                bottom: 0;
                left: 0;
            }

            label {
                font-size: 2rem;
            }
        }
    }
</style>
