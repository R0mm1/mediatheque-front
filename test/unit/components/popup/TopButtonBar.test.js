import {mount} from '@vue/test-utils'
import TopButtonBar, {TopButtonBarElement} from "@/components/popup/TopButtonBar.vue";

describe('TopButtonBar', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(TopButtonBar, {
            sync: false,
            propsData: {
                buttons: [
                    new TopButtonBarElement('fas fa-book', 'main'),
                    new TopButtonBarElement('fas fa-comment', 'social')
                ],
                defaultTab: 'main'
            }
        })
    });

    it('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders correctly after tab changed', async () => {
        wrapper.find('.form_element_button.social').trigger('click');
        await wrapper.vm.$nextTick();
        expect(wrapper.element).toMatchSnapshot();
    });
});
