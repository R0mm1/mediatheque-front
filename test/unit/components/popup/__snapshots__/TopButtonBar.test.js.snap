// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopButtonBar renders correctly 1`] = `
<div
  id="med_popup_topbuttonbar"
>
  <div
    class="form_element main selected form_element_button style_normal"
  >
    <label
      class="element_label fas fa-book"
      for=""
    >
      
            
        
    </label>
     
    <div
      class="element_content hide_content"
    >
      <input
        name=""
        type="button"
      />
    </div>
  </div>
  <div
    class="form_element social form_element_button style_normal"
  >
    <label
      class="element_label fas fa-comment"
      for=""
    >
      
            
        
    </label>
     
    <div
      class="element_content hide_content"
    >
      <input
        name=""
        type="button"
      />
    </div>
  </div>
   
  <div
    class="form_element form_element_button style_normal"
  >
    <label
      class="element_label fas fa-times"
      for=""
    >
      
            
        
    </label>
     
    <div
      class="element_content hide_content"
    >
      <input
        name=""
        type="button"
      />
    </div>
  </div>
</div>
`;

exports[`TopButtonBar renders correctly after tab changed 1`] = `
<div
  id="med_popup_topbuttonbar"
>
  <div
    class="form_element main form_element_button style_normal"
  >
    <label
      class="element_label fas fa-book"
      for=""
    >
      
            
        
    </label>
     
    <div
      class="element_content hide_content"
    >
      <input
        name=""
        type="button"
      />
    </div>
  </div>
  <div
    class="form_element social selected form_element_button style_normal"
  >
    <label
      class="element_label fas fa-comment"
      for=""
    >
      
            
        
    </label>
     
    <div
      class="element_content hide_content"
    >
      <input
        name=""
        type="button"
      />
    </div>
  </div>
   
  <div
    class="form_element form_element_button style_normal"
  >
    <label
      class="element_label fas fa-times"
      for=""
    >
      
            
        
    </label>
     
    <div
      class="element_content hide_content"
    >
      <input
        name=""
        type="button"
      />
    </div>
  </div>
</div>
`;
